programmer/tinyprog/__main__.py

```diff
@@ -326,10 +376,11 @@ def progress(info):
             fpga = TinyProg(active_port, progress)
 
             if args.program is not None:
-                print("    Programming " + str(active_port) + " with " + str(args.program))
+                print("    Programming %s with %s".format(
```

Comment:
@mithro AFAICT you either want:

```
print("    Programming %s with %s" % (FOO, BAR))
```

or 

```
print("    Programming {0} with {1}".format(FOO, BAR))
```

as I don't remember `.format()` taking `%s` style args.  Python 3.6 says no:

```
    Programming %s with %s
    Programming at addr 028000
    Waking up SPI flash
```

Ewen

Reply:
Fixed.